apps/coscrad-frontend/src/components/commands/connections/tag-resource-or-note-form.tsx

```diff
@@ -0,0 +1,87 @@
+import { AggregateType } from '@coscrad/api-interfaces';
+import { isNonEmptyString } from '@coscrad/validation-constraints';
+import {
+    Box,
+    Button,
+    FormControl,
+    FormGroup,
+    MenuItem,
+    Select,
+    Stack,
+    Typography,
+} from '@mui/material';
+import { useState } from 'react';
+import { useLoadableTags } from '../../../store/slices/tagSlice/hooks/use-loadable-tags';
+import { ErrorDisplay } from '../../error-display/error-display';
+import { Loading } from '../../loading';
+import { findOriginalTextItem } from '../../notes/shared/find-original-text-item';
+
+export interface TagResourceOrNoteProps {
+    onSubmitForm: (fsa?: { type: string; payload: Record<string, unknown> }) => void;
+    bindProps: Record<string, unknown>;
+}
+
+export const TagResourceOrNoteForm = ({
+    onSubmitForm,
+    bindProps,
+}: TagResourceOrNoteProps): JSX.Element => {
+    const [selectedTagId, setSelectedTagId] = useState<string>(null);
+
+    const { data, isLoading, errorInfo } = useLoadableTags();
+
+    if (isLoading || data === null) return <Loading />;
+
+    if (errorInfo) return <ErrorDisplay {...errorInfo} />;
```

Comment:
It's so hard to decide whether to use the displayLoadable helper. Sometimes it seems cleaner to just explicitly do the checks without getting lost in layers of wrapping.